test/tests/AelinPool.misc.t.sol

```diff
@@ -0,0 +1,3645 @@
+        }
+
+        vm.expectRevert("nothing left to accept");
+        poolVars.pool.acceptDealTokens(1);
+        vm.stopPrank();
+    }
+
+    function tes_AcceptDealTokens_RevertWhen_MoreThanOpenPeriodShare(
+        uint256 _purchaseDuration,
+        uint256 _poolDuration,
+        uint256 _purchaseTokenAmount,
+        uint256 _purchaseTokenCap,
+        uint256 _proRataRedemptionPeriod,
+        uint256 _openRedemptionPeriod,
+        uint256 _vestingPeriod,
+        uint256 _vestingCliffPeriod,
+        uint256 _underlyingDealTokenTotal,
+        uint256 _holderFundingExpiry,
+        uint256 _purchaseTokenTotalForDeal,
+        uint256 _poolTokenAmount
+    ) public {
+        BoundedVars memory boundedVars = boundVariables(
+            0,
+            _purchaseDuration,
+            _poolDuration,
+            _purchaseTokenAmount,
+            _purchaseTokenCap,
+            _proRataRedemptionPeriod,
+            _openRedemptionPeriod,
+            _vestingPeriod,
+            _vestingCliffPeriod,
+            _underlyingDealTokenTotal,
+            _holderFundingExpiry,
+            _purchaseTokenTotalForDeal
+        );
+
+        PoolVars memory poolVars = getPoolVars(
+            boundedVars.purchaseTokenCap,
+            boundedVars.poolDuration,
+            boundedVars.sponsorFee,
+            boundedVars.purchaseDuration,
+            PoolVarsNftCollection.NONE
+        );
+
+        // Assert
+        vm.startPrank(user1);
+        MockERC20(purchaseToken).approve(address(poolVars.pool), boundedVars.purchaseTokenAmount);
+        poolVars.pool.purchasePoolTokens(boundedVars.purchaseTokenAmount);
+        vm.warp(boundedVars.purchaseDuration + 1);
+
+        poolVars.dealAddress = poolVars.pool.createDeal(
+            address(underlyingDealToken),
+            boundedVars.purchaseTokenTotalForDeal,
+            boundedVars.underlyingDealTokenTotal,
+            boundedVars.vestingPeriod,
+            boundedVars.vestingCliffPeriod,
+            boundedVars.proRataRedemptionPeriod,
+            boundedVars.openRedemptionPeriod,
+            user2,
+            boundedVars.holderFundingExpiry
+        );
+        vm.stopPrank();
+
+        vm.startPrank(user2);
+        deal(address(underlyingDealToken), address(user2), type(uint256).max);
+        MockERC20(address(underlyingDealToken)).approve(address(poolVars.dealAddress), boundedVars.underlyingDealTokenTotal);
+        AelinDeal(poolVars.dealAddress).depositUnderlying(boundedVars.underlyingDealTokenTotal);
+        vm.stopPrank();
+
+        vm.startPrank(user1);
+        poolVars.pool.acceptDealTokens(_poolTokenAmount);
+        vm.warp(block.timestamp + boundedVars.proRataRedemptionPeriod);
+
+        vm.expectRevert("accepting more than share");
+        poolVars.pool.acceptDealTokens(_poolTokenAmount);
+        vm.stopPrank();
+    }
+
+    function testFuzz_AcceptDealTokens_Pool(
```

Comment:
I think for every Accept/AcceptMax function here we need to make sure users got their NFT (MockERC721/AelinVestingToken) and check if the vesting schedule associated with it makes sense